### ISSUE TITLE
InputMask: server-side validation of mask attribute skips required-validation

### ISSUE THREAD

**Issue by fanste** [primefaces/primefaces#4289]
## 1) Environment
- PrimeFaces version: 6.3 SNAPSHOT as of 28.11.2018 09:30 UTC+1
- Application server + version: Wildfly 10.1, JSF Mojarra 2.2.13.SP1 20160303-1204
- Affected browsers: All

## 2) Expected behavior
Input mask with `required = true` should validate and set `validationFailed = true` when nothing or something invalid (not matching the mask) was inserted into the field.

## 3) Actual behavior
Submitted value of the input mask will be set to `null` and therefore the whole validation is skipped. UIInput#validate(FacesContext) contains the code
```
// Submitted value == null means "the component was not submitted
// at all".  
Object submittedValue = getSubmittedValue();
if (submittedValue == null) {
    return;
}
```

This behavior was introduced by  #3234 / #3237

## 4) Steps to reproduce
Add an <p:inputMask requiired="true" ... /> to your page and insert nothing / something invalid according to the mask. Submit the field. The input is accepted as valid and the whole JSF lifecycle is executed including invoke-application, etc

## 5) Sample XHTML
```
<h:form>
    <p:inputMask value="#{myBean.maskedValue}" mask="99:99" required="true" />
    <p:commandButton value="execute" process="@form" update="@form" actionListener="#{myBean.doProcessMaskedValue}" />
</h:form>
```

## 6) Sample bean
```
@ViewScoped
@Named
public class MyBean implements Serializable {

    private String _maskedValue;

    public void doProcessMaskedValue() {
        System.out.prinln("Masked value should not be NULL, current value is = " + _maskedValue);
    }

    public String getMaskedValue() {
        return _maskedValue;
    }

    public void setMaskedValue(String maskedValue) {
        _maskedValue = maskedValue;
    }
}
```

..

**Comment by fanste** [comment]
There would be multiple solutions how to fix that:
- Set `isValid(false)` and set a message for the component
- Override some of the validation methods, but ensure that it does not conflict with the converter (own validation code has to be done before the converter is invoked)
- use the converter (getConvertedValue) to throw an exception when the mask does not match

I don't know what would be the best / correct solution

**Comment by tandraschko** [comment]
couldn't we just set it to "", so the original coverter logic will be executed?
All of your solutions are quite hacky, we should restore the behavior that our regex validation is just a "additional feature" and the validators are still executed.

**Comment by fanste** [comment]
Well.. that would be the best solution. That's to simple to come into my mind :D

**Comment by fanste** [comment]
PR #4298

**Comment by tandraschko** [comment]
Thanks for the PR!

**Comment by fanste** [comment]
New PR  #4316.

Maybe I've tested the wrong code but the bug existed again. The `isEmpty`-check was at the wrong position which totally skipped the setter of the submitted value..
